core/src/main/java/org/pentaho/di/core/Result.java

```diff
@@ -0,0 +1,935 @@
+
+  /** The number of errors during the transformation or job */
+  private long nrErrors;
+
+  /** The number of lines input. */
+  private long nrLinesInput;
+
+  /** The number of lines output. */
+  private long nrLinesOutput;
+
+  /** The number of lines updated. */
+  private long nrLinesUpdated;
+
+  /** The number of lines read. */
+  private long nrLinesRead;
+
+  /** The number of lines written. */
+  private long nrLinesWritten;
+
+  /** The number of lines deleted. */
+  private long nrLinesDeleted;
+
+  /** The number of files retrieved. */
+  private long nrFilesRetrieved;
+
+  /** The result of the job or transformation, true if successful, false otherwise. */
+  private boolean result;
+
+  /** The entry number. */
+  private long entryNr;
+
+  /** The exit status. */
+  private int exitStatus;
+
+  /** The rows. */
+  private List<RowMetaAndData> rows;
+
+  /** The result files. */
+  private Map<String, ResultFile> resultFiles;
+
+  /** Whether the job or transformation was stopped. */
+  public boolean stopped;
+
+  /** The number of lines rejected. */
+  private long nrLinesRejected;
+
+  /** The log channel id. */
+  private String logChannelId;
+
+  /** The log text. */
+  private String logText;
+
+  /**
+   * safe stop.
+   */
+  private boolean safeStop;
+
+  /**
+   * Elapsed time of the ETL execution in milliseconds
+   */
+  private long elapsedTimeMillis;
+
+  /**
+   * Unique identifier of an ETL execution, should one ever care to declare one such
+   */
+  private String executionId;
+
+  /**
+   * Instantiates a new Result object, setting default values for all members
+   */
+  public Result() {
+    nrErrors = 0L;
+    nrLinesInput = 0L;
+    nrLinesOutput = 0L;
+    nrLinesUpdated = 0L;
+    nrLinesRead = 0L;
+    nrLinesWritten = 0L;
+    result = false;
+
+    exitStatus = 0;
+    rows = new ArrayList<RowMetaAndData>();
+    resultFiles = new ConcurrentHashMap<String, ResultFile>();
+
+    stopped = false;
+    entryNr = 0;
+  }
+
+  /**
+   * Instantiates a new Result object, setting default values for all members and the entry number
+   *
+   * @param nr
+   *          the entry number for the Result
+   */
+  public Result( int nr ) {
+    this();
+    this.entryNr = nr;
+  }
+
+  /**
+   * Performs a semi-deep copy/clone but does not clone the rows from the Result
+   *
+   * @return An almost-clone of the Result, minus the rows
+   */
+  public Result lightClone() {
+    // This light-weight clone doesn't clone rows
+    try {
+      Result result = (Result) super.clone();
+      result.setRows( null );
+      if ( resultFiles != null ) {
+        Map<String, ResultFile> clonedFiles = new ConcurrentHashMap<String, ResultFile>();
+        Collection<ResultFile> files = resultFiles.values();
+        for ( ResultFile file : files ) {
+          clonedFiles.put( file.getFile().toString(), file.clone() );
+        }
+        result.setResultFiles( clonedFiles );
+      }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 187:194, 224:231.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.